### PR TITLE
feat: terraform handles multiple environement

### PR DESCRIPTION
**Pull request by EdouardDem**
# Refactor environments management

Infrastructure becomes much more customizable from the terraform app.
Containers embed their own default config.
Configuration from the app is now typed and lighter.
Scaling and image tags are configurable from the app, and can be overridden for each env (production, staging, etc.).

# Samples

Terraform app config example https://github.com/tractr/stack/blob/32626787b5c40798fbb07ba882096cdb8a523443/apps/terraform/src/configs/apps.config.ts

Environments config example
https://github.com/tractr/stack/blob/32626787b5c40798fbb07ba882096cdb8a523443/apps/terraform/src/configs/environments.config.ts

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -34,15 +34,25 @@ module.exports = {
     '<rootDir>/libs/nestjs/mailer',
     '<rootDir>/libs/nestjs/testing',
     '<rootDir>/libs/common',
-    '<rootDir>/libs/terraform/aws-component',
+    '<rootDir>/libs/terraform/components/aws-component',
```
Comment (floross) [inline jest.config.js]: The folder name should not be suffix by `-component` if scoped in a folder ?

File: libs/terraform/services/ecs-services/src/lib/interfaces/container.interface.ts [hunk 2]
```diff
@@ -60,10 +47,30 @@ export interface MountPoint {
...
   name: string;
+}
+
+export interface ContainerPublicConfig {
+  imageTag: string;
   cpu?: number;
   memory?: number;
-  secrets?: (string | SecretMap)[];
-  environments?: (Environment | EnvironmentCb)[];
+  environments?: Record<string, EnvironmentOrSecretValue>;
```
Comment (floross) [inline libs/terraform/services/ecs-services/src/lib/interfaces/container.interface.ts]: Is it possible to have default value for `type` to `env` ? It should remove some surplus. Else we could keep the secrets key and only have a key/value hash

**Review by floross — DISMISSED** [review]
We should keep the scope in the package name like you did for the folder names:

`@tractr/terraform-reverse-proxy-service` => `@tractr/terraform-service-reverse-proxy`
`libs/terraform/services/reverse-proxy-service` => `libs/terraform/services/reverse-proxy`

What do you think ?

File: jest.config.js (hunk 1 above)
Comment (EdouardDem) [reply jest.config.js]: I have done this for `-group` and `-service` also

File: libs/terraform/services/ecs-services/src/lib/interfaces/container.interface.ts (hunk 2 above)
Comment (EdouardDem) [reply libs/terraform/services/ecs-services/src/lib/interfaces/container.interface.ts]: Done. Check this file to get all configuration available: https://github.com/tractr/stack/blob/f241e7ad0b75e1dec2c30a6c1ed4e9969fee2473/libs/terraform/services/ecs-services/README.md

**Review by maxmousse — APPROVED** [review]

**Pull request merged**